dipy/align/_public.py

```diff
@@ -0,0 +1,833 @@
+    if level_iters is None:
+        level_iters = [10000, 1000, 100]
+    if sigmas is None:
+        sigmas = [3, 1, 0.0]
+    if factors is None:
+        factors = [4, 2, 1]
+
+    static, static_affine, moving, moving_affine, starting_affine = \
+        _handle_pipeline_inputs(moving, static,
+                                moving_affine=moving_affine,
+                                static_affine=static_affine,
+                                starting_affine=starting_affine)
+
+    # Define the Affine registration object we'll use with the chosen metric:
+    use_metric = affine_metric_dict[metric](
+                        nbins=nbins,
+                        sampling_proportion=sampling_proportion)
+
+    affreg = AffineRegistration(metric=use_metric,
+                                level_iters=level_iters,
+                                sigmas=sigmas,
+                                factors=factors)
+
+    # Go through the selected transformation:
+    for func in pipeline:
+        starting_affine = func(moving, static,
+                               static_affine=static_affine,
+                               moving_affine=moving_affine,
+                               starting_affine=starting_affine,
+                               reg=affreg)
+
+    # After doing all that, resample once at the end:
+    affine_map = AffineMap(starting_affine,
+                           static.shape, static_affine,
+                           moving.shape, moving_affine)
+
+    resampled = affine_map.transform(moving)
+
+    return resampled, starting_affine
+
+
+def register_series(series, ref, pipeline=None, series_affine=None,
+                    ref_affine=None):
+    """Register a series to a reference image.
+
+    Parameters
+    ----------
+    series : 4D array or nib.Nifti1Image class instance or str
+        The data is 4D with the last dimension separating different 3D volumes
+
+    ref : int or 3D array or nib.Nifti1Image class instance or str
+        If this is an int, this is the index of the reference image within the
+        series. Otherwise it is an array of data to register with (associated
+        with a `ref_affine` required) or a nifti img or full path to a file
+        containing one.
+
+    pipeline : sequence, optional
+        Sequence of transforms to do for each volume in the series.
+        Default: (executed from left to right):
+        `[c_of_mass, translation, rigid, affine]`
+
+    series_affine, ref_affine : 4x4 arrays, optional.
+        The affine. If provided, this input will over-ride the affine provided
+        together with the nifti img or file.
+
+    Returns
+    -------
+    xformed, affines : 4D array with transformed data and a (4,4,n) array
+    with 4x4 matrices associated with each of the volumes of the input moving
+    data that was used to transform it into register with the static data.
+    """
+    if pipeline is None:
+        [c_of_mass, translation, rigid, affine]
```

Comment:
same as above

Reply:
I just also noticed that this was not even being assigned 🤣 

Done in 833e508c2